packages/neo-one-smart-contract-lib/src/crowdsale/CrowdsaleContract.ts

```diff
@@ -0,0 +1,93 @@
+import { Address, constant, createEventNotifier, Fixed, SmartContract } from '@neo-one/smart-contract';
+import { Ownable } from '../ownership/Ownable';
+
+const notifyTokensPurchased = createEventNotifier<Address, Address, Fixed<8>, Fixed<8>>(
+  'tokens_purchased',
+  'purchaser',
+  'beneficiary',
+  'NEO',
+  'qty tokens',
+);
+
+export function CrowdsaleContract() {
+  abstract class CrowdsaleContractClass extends Ownable(SmartContract) {
+    protected mutableNeoRaised: Fixed<8> = 0;
+    protected mutableTotalSupply: Fixed<8> = 0;
+    private readonly _token: Address = this.initialCrowdsaleToken();
```

Comment:
something something I don't think we want variables prefixed with `_` . Might be a dicarlo thing.

Reply:
@dicarlo2  is there an appropriate time to use _ at the beginning of a variable name?
For some, the convention is:
"private variables are started with _ , nothing more than that"